modules/darwin/src/main/ts/ephox/darwin/selection/SelectionTypes.ts

```diff
@@ -0,0 +1,33 @@
+import { Adt } from '@ephox/katamari';
+import { SugarElement } from '@ephox/sugar';
+
+export interface SelectionType {
+  fold: <T>(
+    none: () => T,
+    multiple: (elements: SugarElement<Element>[]) => T,
```

Comment:
Should this be `SugarElement<HTMLTableCellElement>[]` instead (and likewise below)? As I believe this is only intended to work with tables, but I can't recall if Darwin is meant to be used elsewhere?

Reply:
Did this and it worked, but when I tried to feed it through more places things exploded so have only done the minimum.